### DISCUSSION TITLE
How to mark "bootstrap/startup" class for OnionArchitecture?

### DISCUSSION THREAD

**Discussion by tedyoung** [TNG/ArchUnit#916] [General]
In my Hexagon Architected Spring Boot app, I have the main class (`BlackjackGameApplication`), with the `@SpringBootApplication` annotation that does the bootstrap configuration of the application. It does this by instantiating objects in the App Service Layer and the Domain. However, the bootstrap class itself is in the root package (`dev.ted.blackjack`). There doesn't appear to be a way to tell ArchUnit's `Architectures.onionArchitecture()` that this class can reference anything (it's just transient configuration, so it is OK).

I can exclude that class from being imported altogether (using `withImportOption`), but would be nice if I could specify it as part of the Architectures rule, e.g., `bootstrapClass("dev.ted.BlackjackGameApplication")`.

Am I missing something? If not, I can open an issue.

**Comment by codecholeric** [comment]
You can use an ignore for that:

```
onionArchitecture()
  ...
  .ignoreDependency(JavaClass.Predicates.belongToAnyOf(BlackjackGameApplication.class), alwaysTrue())
```

My problem with `bootstrapClass(..)` is, that I'm not sure this is a generic concept for all cases. It makes sense with Spring Boot, but I'm not sure if every project has a bootstrap class outside of the architecture. So I would rather stick to the generic `ignoreDependency` for that. But I might be convinced otherwise, if the community agrees that there should always be a "bootstrap" class as a explicitly defined component of the Onion architecture :wink:

**Reply by jonas-grgt** [reply]
The `ignoreDependency` solution does not seem to work for me. Yet, adding it at the end does:
```java
onionArchitecture()
...
.ensureAllClassesAreContainedInArchitectureIgnoring(JavaClass.Predicates.type(BlackjackGameApplication.class));
```
Is this a bug?

**Reply by codecholeric** [reply]
Could this be related to https://github.com/TNG/ArchUnit/pull/1226 ?
